src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { getUser } from '../../api';
+import { useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const targetTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function setTargetUser() {
+    if (!targetTodo) {
+      setUser(null);
+
+      return;
+    }
+
+    setIsLoading(true);
+    const newUser = await getUser(targetTodo.userId);
+
+    try {
+      setUser(newUser);
+    } catch {
+      throw new Error();
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  useEffect(() => {
+    setTargetUser();
+  }, [targetTodo]);
```

Comment:
Same with the function. Wrap it into the useCallback and use it in the dependency array
```suggestion
  useEffect(() => {
    setTargetUser();
  }, [setTargetUser]);
```